core/auth/src/main/java/software/amazon/awssdk/auth/credentials/internal/SsoCredentialsProfileProperties.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.auth.credentials.internal;
+
+import software.amazon.awssdk.annotations.SdkInternalApi;
+
+/**
+ * A container for SSO credential properties, these properties can be loaded from the profile in default.
+ */
+@SdkInternalApi
+public class SsoCredentialsProfileProperties {
```

Comment:
Should this be named with "Profile" since it's not necessarily related to a profile file? I'd honestly rather just pass a `Profile` to a `ProfileCredentialsProviderFactory` instead of a `SsoCredentialsProfileProperties` to a `SsoCredentialsProviderFactory`, and not make this specific to SSO. If we had done that for the web identity token file, we wouldn't have to create yet another protected API here.

Reply:
It's not necessary to include "Profile" in the name but I can't come up with a better name. Do you have any good idea about how to name this? I named it with "Profile" because it takes the configurations from the profile file and those properties are all related to the SSO credentials provider.

Reply:
What do you think about my alternate suggestion - making it more generic and just passing the profile through?

Reply:
I understand your point of making the factory generic so we don't need to add more and more factories as protected apis here. But I think our purpose of using `ProfileCredentialsUtils` to create different credentials providers from the profile properties is just to make the reading properties part generic. If we pass a `Profile` directly to the `ProfileCredentialsProviderFactory`, then we would need to add a series of reading properties and validating properties stuffs in the factory itself. Isn't that conflicting with our original design? If you do think creating less protected apis is worthy then do you think we should make it independent from the current `ProfileCredentialsUtils`?

Reply:
To make sure we're on the same page, I was picturing just changing the current `SsoCredentialsProviderFactory` to `ProfileCredentialsProviderFactory`, and then we'd have an `SsoProfileCredentialsProviderFactory` that implements that interface. The responsibilities of the calling class and the factory itself would remain the same.

The `SsoProfileCredentialsProviderFactory` would indeed read the properties from the profile, but that's just calling `profile.property(foo)` instead of `ssoCredentialsProfileProperties.foo()`. Would there have to be more than that?

Reply:
I see, sorry I misunderstood your suggestion, will do the change.